As a Game Control, I wish to be able to hide games

lebaphi opened the pull request:
Fixes #900

philb89 committed:
As a Game Control, I wish to be able to hide games

philb89 committed:
fix lint

philb89 committed:
fix build

IanMayo committed:
Merge branch 'develop' into 900-allow-hide-game

IanMayo commented:
Thanks for this @lebaphi :-)

I've put a note in Slack about future PRs (not this one) - about helping my by trying to keep them focussed on the new feature - and include bulk linter/tidying for separate PRs.

IanMayo reviewed (changes requested):
Aah, I do have a change request, please @lebaphi 

In Game Admin, I reduced the games down to just one.

When I refreshed the Player UI, just one game was shown.

But, we've introduced the logic where if just one game is present, the UI _loads_ that game (#873).

Could you investigate what that isn't happening, please?

philb89 committed:
make PR clean to review

lebaphi commented:
> Aah, I do have a change request, please @lebaphi
> 
> In Game Admin, I reduced the games down to just one.
> 
> When I refreshed the Player UI, just one game was shown.
> 
> But, we've introduced the logic where if just one game is present, the UI _loads_ that game (#873).
> 
> Could you investigate what that isn't happening, please?

okay @IanMayo

lebaphi commented:
@IanMayo I removed some code refactoring that does not relate to this feature. So it's easier for you to review. We may have another PR to do the code refactor or clean code.

philb89 committed:
auto select the Wargame if the list has one

lebaphi commented:
> > Aah, I do have a change request, please @lebaphi
> > In Game Admin, I reduced the games down to just one.
> > When I refreshed the Player UI, just one game was shown.
> > But, we've introduced the logic where if just one game is present, the UI _loads_ that game (#873).
> > Could you investigate what that isn't happening, please?
> 
> okay @IanMayo

Fixed @IanMayo

IanMayo commented on packages/client/src/Views/PlayerUi/PlayerUi.tsx:
```diff
@@ -78,6 +79,10 @@ const PlayerUi = ({ gameInfo, wargame, messageTypes, checkPasswordFail, loadData
 
 
   const handleCheckPassword = (pass: string): void => {
+    if (currentWargame.startsWith(hiddenPrefix)) {
```
Hello @lebaphi - this test shouldn't fail, should it?

The hidden wargame shouldn't be listed, so it can't be present as `currentWargame`, can it.

I _guess_ this is handling some edge case.  Based on it being an edge case, I suggest the notification at line 21 below be changed to:
`Error: hidden wargame should not be available. Not opening.`

lebaphi replied:
Hi @IanMayo, the hidden wargame shouldn't be listed, right. This handles the case:
- in the client-side, the user loads the list of 3 games
- at this time on the server-side, the admin hide 1 of 3 games
- in the client-side, the user does not reload the page and chooses the hidden game then press enter.

=> we should handle this case and show an error message. I will update the error message.

philb89 committed:
update error message
